### DISCUSSION TITLE
SWR with useSelector, useEffect with depedency

### DISCUSSION THREAD

**Discussion by denisonfer** [vercel/swr#818] [Q&A]
hello folks

Does anyone know how to redo the code snippet below using SWR?
```
const dependency= useSelector(state...)

useEffect(() => {
   call api...
  }, [dependency]);

````

**Comment by ath92** [comment]
I believe it'd look something like

```
const dependency= useSelector(state...)
const { data } = useSWR(["endpointKey", dependency]);
```

You'd have to write a fetcher function that uses the dependency in a way that makes sense for your use case, but it's hard to know what that'd look like without more information.
